File: android/app/src/main/java/com/mendoi/kotodama/data/Affirmation.kt
package com.mendoi.kotodama.data

data class Affirmation(
    val id: String,
    val text: String,
    val category: String,
    val customCategoryName: String? = null,
    val morningEnabled: Boolean = true,
    val eveningEnabled: Boolean = false,
    val orderIndex: Int = 0,
)

/// iOS と同じデフォルト 20件 seed
object DefaultAffirmations {
    val seed: List<Affirmation> = listOf(
        Affirmation("c1", "私は自分の人生を自分でデザインしている", "selfAffirm", "自信"),
        Affirmation("c2", "私は幸せを選んでいる", "selfAffirm", "自信"),
        Affirmation("c3", "私には夢を叶える力がある", "selfAffirm", "自信"),
        Affirmation("c4", "私は自分のベストバージョンに近づいている", "selfAffirm", "自信"),
        Affirmation("g1", "私はお金を簡単に、素早く引き寄せている", "goal", "豊かさ"),
        Affirmation("g2", "私は富を引き寄せる磁石だ", "goal", "豊かさ"),
        Affirmation("g3", "私は豊かさの中で生きている", "goal", "豊かさ"),
        Affirmation("g4", "私はより豊かになるチャンスに恵まれている", "goal", "豊かさ"),
        Affirmation("h1", "今日の私は、エネルギーに満ちあふれていて何でもできる", "habit", "生産性"),
        Affirmation("h2", "私は生産的に仕事をしている", "habit", "生産性"),
        Affirmation("h3", "私は計画を実行できている", "habit", "生産性"),
        Affirmation("h4", "私の頭の中も周りの環境もよく整理されていて、物事を着実に進めることができている", "habit", "生産性"),
        Affirmation("v1", "私は人の愛を受け入れて、人に愛を与えている", "values", "人間関係"),
        Affirmation("v2", "私は家族や友人といった素晴らしい人間関係に恵まれている", "values", "人間関係"),
        Affirmation("v3", "私は人を大切にするし、周りの人も私を大切にしてくれる", "values", "人間関係"),
        Affirmation("v4", "私は愛を受け入れる準備があるし、愛に値する人間だ", "values", "人間関係"),
        Affirmation("h5", "私は自分の体にいいものだけを、必要なときにだけ食べる", "values", "健康"),
        Affirmation("h6", "私は健康な体を持っている", "values", "健康"),
        Affirmation("h7", "私は理想通りの、健康で、強い体を持つ自分に近づいている", "values", "健康"),
        Affirmation("h8", "私は自分の体を愛し、尊敬している", "values", "健康"),
    )
}

File: AFFARMENTNEMO/Models/Affirmation.swift
//
//  Affirmation.swift
//  3モード(内部分類のみ、UIには出さない) + カテゴリ管理
//  仕様: 02_開発仕様書_v3 §2.2 / 04_アファメーション科学的根拠 / 05_UXフロー §0.2
//

import Foundation
import SwiftData

@Model
final class Affirmation {
    @Attribute(.unique) var id: UUID
    var text: String
    var internalMode: String     // "affirmation" | "value" | "ifThen"  (UIには出さない)
    var templateUsed: String?    // 「なりたい自分」「習慣のキッカケ」など (分析用)
    var category: String         // "goal" | "habit" | "selfAffirm" | "plan" | "concern" | "values" | "custom"
    var customCategoryName: String?

    // モード2 (価値アファメ) 用 - MVPでは未使用、後方互換のため保持
    var coreValues: [String]
    var valueReason: String?

    // モード3 (If-Thenプラン) 用
    var ifTrigger: String?
    var thenAction: String?

    // 通知スケジュール
    var morningEnabled: Bool
    var eveningEnabled: Bool

    // 統計
    var readCount: Int
    var lastReadAt: Date?
    var currentStreakDays: Int
    var longestStreakDays: Int

    // メタデータ
    var createdAt: Date
    var updatedAt: Date
    var orderIndex: Int
    var isActive: Bool

    // 録音 (ユーザー要望: 自分の声で再生)
    /// Documents/recordings/ 配下のファイル名 (m4a)。nil = 未録音。
    var recordingFileName: String?

    // ホームのルーティンに含めるか (\u30db\u30fc\u30e0\u8aad\u307f\u4e0a\u3052\u30bb\u30c3\u30c8 \u30aa\u30f3/\u30aa\u30d5)
    var includeInRoutine: Bool = true

    init(
        text: String,
        internalMode: String = "affirmation",
        templateUsed: String? = nil,
        category: String = "custom",
        customCategoryName: String? = nil,
        coreValues: [String] = [],
        valueReason: String? = nil,
        ifTrigger: String? = nil,
        thenAction: String? = nil,
        morningEnabled: Bool = true,
        eveningEnabled: Bool = false,
        orderIndex: Int = 0
    ) {
        self.id = UUID()
        self.text = text
        self.internalMode = internalMode
        self.templateUsed = templateUsed
        self.category = category
        self.customCategoryName = customCategoryName
        self.coreValues = coreValues
        self.valueReason = valueReason
        self.ifTrigger = ifTrigger
        self.thenAction = thenAction
        self.morningEnabled = morningEnabled
        self.eveningEnabled = eveningEnabled
        self.readCount = 0
        self.lastReadAt = nil
        self.currentStreakDays = 0
        self.longestStreakDays = 0
        self.createdAt = Date()
        self.updatedAt = Date()
        self.orderIndex = orderIndex
        self.isActive = true
    }
}
